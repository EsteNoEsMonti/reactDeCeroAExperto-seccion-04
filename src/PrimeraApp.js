import React from 'react';
import PropTypes from 'prop-types';

// FC
const PrimeraApp = ({ saludo, subtitulo }) => {

  return (
    <>
      <h1>{saludo}xdxdxd</h1>
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> null es el replace y 3 el espacio */}
      <p>{subtitulo}</p>
    </>
  );
};

//aqui podemos definir el tipo de nuestras propiedades
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
};

export default PrimeraApp;
