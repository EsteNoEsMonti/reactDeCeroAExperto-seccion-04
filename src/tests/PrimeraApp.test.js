import React from 'react';
import { shallow } from 'enzyme';
// import { render } from '@testing-library/react';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp/>', () => {

  // test('Debe de mostrar el mensaje "Que hay de nuevo viejo?"', () => {

  //   // debo renderizar el componente y evaluar sobre el mismo
  //   const saludo = 'Que hay de nuevo viejo?';

  //   //wrapper es el producto renderizado de lo que quiero mostrar
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />); //recibe un componente, que es lo que quiero renderizar

  //   expect(getByText(saludo)).toBeInTheDocument();

  // });

  test('Debe de mostrar <PrimeraApp/> correctamente', () => {

    const saludo = 'Que hay de nuevo viejo?';

    //el hsallow es similar al render pero nos va a dar otras opciones como simular clicks etc
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar el subtitutlo enviado por props', () => {

    const saludo = 'Que hay de nuevo viejo?';
    const subtitulo = 'this is a fucking subtitle';

    //el hsallow es similar al render pero nos va a dar otras opciones como simular clicks etc
    const wrapper = shallow(
      <PrimeraApp
        saludo={saludo}
        subtitulo={subtitulo}
      />
    );

    const textoParrafo = wrapper.find('p').text(); //puede buscar por #id .clases o elementos h1 h2 p

    expect(subtitulo).toBe(textoParrafo);
    
  });


});