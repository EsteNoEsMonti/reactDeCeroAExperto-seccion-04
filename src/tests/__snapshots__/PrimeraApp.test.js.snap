// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe de mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Que hay de nuevo viejo?
    xdxdxd
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp/> Debe de mostrar el subtitutlo enviado por props 1`] = `
<Fragment>
  <h1>
    Que hay de nuevo viejo?
    xdxdxd
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
